### DISCUSSION TITLE
Support system font fallbacks

### DISCUSSION THREAD

**Discussion by jackpot51** [pop-os/cosmic-text#14] [Ideas]
Determine if and how to support system fallbacks (e.g. fontconfig on Linux)

**Comment by xorgy** [comment]
Discussion on the Discord reveals:
FontConfig users don't want or need perfect emulation of libfontconfig, but fallback should ideally respect common fontconfig rules from FC config files files in the standard locations, especially around generic family names and deliberate substitutions.

**Comment by genusistimelord** [comment]
I believe we should find a way to parse the files to avoid using font config directly and just get the font changes. currently this does exist
https://github.com/fschutt/rust-fontconfig  which did just that.

I would probably make it so
If the system has the font config file then load and use it first.
If not load the default fallbacks
if all else fails and no font was loaded Error other wise if fonts did get loaded but no support for the symbol then return the normal
square symbol for not supported symbol.

**Comment by vikigenius** [comment]
At least for the "if" part, I will say that a lot of applications on Linux including browsers do support fontconfig especially when it comes to missing characters, even if they do have their own preferences.

For example here: https://neugierig.org/software/chromium/fonts/

> Because web pages themselves (via CSS) provide their own font fallback preferences, Chrome disregards the font fallback provided by fontconfig except when the fonts requested by the page don't provide the characters needed.

So I agree with @xorgy in that at the very least we can parse and respect the user's FC config files as far as the docs specify.

**Comment by xorgy** [comment]
For reference, the standard locations are as follows (transcribed from `FONTS-CONF(5)`): 
 - `/etc/fonts/fonts.conf`
 - `/etc/fonts/conf.d` (directory of fonts.conf compatible files)
 - `$XDG_CONFIG_HOME/fontconfig/conf.d` (directory of fonts.conf compatible files)
 - `$XDG_CONFIG_HOME/fontconfig/fonts.conf`
 - `~/.fonts.conf.d` (directory of fonts.conf compatible files)
 - `~/.fonts.conf`
